src/routers/handlers/enter_your_details/index.ts

```diff
@@ -33,7 +34,15 @@ export class EnterYourDetailsHandler extends GenericHandler<EnterYourDetailsView
     public getViewData(req: Request): EnterYourDetailsViewData {
         const baseViewData = super.getViewData(req);
         // retrieve details using session key
-        const details = getPresenterAccountDetails(req);
+        let details = getPresenterAccountDetails(req);
+        if (details.userId !== undefined) {
+            const generatedDetails =  generateUserDetails(req, {} as Address);
+            details = { ...details,
+                userId: generatedDetails.userId,
+                name: generatedDetails.name,
+                email: generatedDetails.email,
+                createdDate: generatedDetails.createdDate };
+        }
```

Comment:
There are two problems with this.
1. This code doesn't work and fails to render tthe enter you details page due to the same error that I mentioned in my previous review. The details are not initially in the session. Calling `getPresenterAccountDetails` will result in an error being thrown. You much check if they are present before hand and populate them if neccessary.
2. You cannot simply cast an empty to object to an `Address` type as it is not an address type. It should be retireved from the request body and checked to ensure it is the correct type and the error should be handled if not.